src/cache.ts

```diff
@@ -139,7 +139,8 @@ export class CMakeCache {
       if (line.startsWith('//')) {
         docs_acc += /^\/\/(.*)/.exec(line)![1] + ' ';
       } else {
-        const match = /^(.*?):(.*?)=(.*)/.exec(line);
+        const entry_pattern = line.startsWith('\"') ? /^\"(.*?)\":(.*?)=(.*)/ : /^(.*?):(.*?)=(.*)/;
```

Comment:
Usually, I would prefer to use regex syntax to differentiate between the two cases:
```
^("(.*?)"|(.*?)):([^:]*?)=(.*)
```
Then you can use just one pattern.  (you'd have to update the unpacking on line 148 though since there's an extra capture group in this one).

Reply:
Hi @bobbrow - thanks for the feedback.  I like what you are suggesting, but I have not been able to  get it working.  I plugged it at https://regexr.com/ as well and can reproduce the issue.  For whatever reason, the extra parens you added for the either-or aren't liked.  

I tried this, and it works: 
`(^"(.*?)"|(.*?)):([^:]*?)=(.*)`
But, I don't want the carat included in grouping.

I got this to work as well:
`"(.*?)"|(.*?):([^:]*?)=(.*)`
But, this makes the capturing a little crazy (could be in the first group or second group)

Is there something I'm missing that you can see?  I'm pretty decent with regex, but this one has me a little surprised.

Reply:
Ignore my previous comment - it was too late when I was looking at this.  For the record, I deleted the comment because I couldn't seem to add a new comment otherwise on this thread.

Reply:
Nno worries.  I was just going to comment that I tested my regex on https://regex101.com/ and it worked there.  I'm not familiar with the site you shared so I don't know what's going on there.

Reply:
With your current set of changes though, I think you need to update line 147 to unpack the values correctly.  I believe it would be (add a comma):
```ts
const [, , name, typename, valuestr] = match;
```

Reply:
Yep, already done.. I used the following to unpack:
`const [, name, , , typename, valuestr] = match;`
But, I am realizing that there is one caveat - the name for the quote case that I want is in group 2, but group 3 (or 1) for the non-quoted case. I'll need extra code for that.
Or, conversely, I can use group 1 for both, but manually need to strip the quotes if they are there.

Do you have a preference on one of those two ways?  Or perhaps another idea? 

See: https://regex101.com/r/djovn6/1

Reply:
Ok, I think this will work:
`^(?|"(.*?)"|(.*?)):([^:]*?)=(.*)`
Learned something new today! :)

Reply:
Hmm.. Spoke to soon, I think. Seems that typescript doesn't support this?

Reply:
Another alternative - I kind of like this:
```
        const match = /^("(.*?)"|(.*?)):([^:]*?)=(.*)/.exec(line);
        if (!match) {
          rollbar.error(localize('failed.to.read.line.from.cmake.cache.file', 'Failed to read a line from a CMake cache file {0}', line));
          continue;
        }
        const filtered = match.filter(element => element != undefined)
        const [, , name, typename, valuestr] = filtered;
```

I'm not very well versed with javascript or typescript (I'm a C++ & Python guy).  So the above may not not be very clear to someone with more experience in typescript :)

Reply:
I think that's what I suggested earlier. 😉 But your PR currently has the commas in the wrong place for unpacking the result.  Update it to what you typed above and I think it's good to go.

I would also recommend adding another test:
const str = "\"QUOTED\":STRING=value";

Reply:
And yes, I realize now that I missed that regex101 was filtering the undefined results automatically, so my original proposed solution was missing that.

Reply:
Just update the PR with your proposed solution and we'll approve it.

Reply:
I had to fiddle with the filter line a little to get it past the linter, but it's compiling and all tests are passing on linux (including the new one I added based on your suggestion).
I expect Mac will pass on travis as well, so I think this PR will be good to go!

Reply:
Sorry, my fault... It should now be what you expect.